extensions/lifecycle/objectProcessor/LifecycleObjectTransitionProcessor.js

```diff
@@ -53,12 +55,14 @@ class LifecycleObjectTransitionProcessor extends LifecycleObjectProcessor {
      * @return {undefined}
      */
     start(done) {
-        super.start(err => {
-            if (err) {
-                return done(err);
-            }
-            return this.setupProducer(done);
-        });
+        async.waterfall([
+            next => super.start(next),
+            next => {
+                this._gcProducer = new GarbageCollectorProducer();
```

Comment:
this is done using waterfall because the start method needs to complete successfully before setting up the producers ( this is what was done initially )